Library/Homebrew/command.rb

```diff
@@ -0,0 +1,133 @@
+module Homebrew
+  class Command
+    attr_accessor :command_name, :description
+    attr_reader :valid_options, :help_output, :man_output
+
+    def initialize_variables
+      @valid_options = []
+      @argv_tokens = ARGV.dup.uniq
+    end
+
+    def options(&block)
+      initialize_variables
+      instance_eval(&block)
+      generate_documentation
+    end
+
+    def option(option_name, **option_hash, &block)
+      option_hash[:option_name] = "--#{option_name}"
+      option_name = option_hash[:option_name]
+      option_hash[:child_option_names] = []
+      if @parent_option_name.nil?
+        option_hash[:is_root_option] = true
```

Comment:
Nit: shorten `is_root_option` down to just `root` or `root_opt`. The `is` is implied by its boolean-ness.